### PR TITLE
vim-patch:9.1.0196: filetype: support for gnuplot files is lacking

### PR DESCRIPTION
**Pull request by clason**
Problem:  filetype: support for gnuplot files is lacking
Solution: Also detect *.gnuplot files
          (RobbiZ98)

closes: vim/vim#14243

https://github.com/vim/vim/commit/3a6bd0c5c743bf69d2e8af4c8b3c6b2cb5f3631a

Co-authored-by: RobbiZ98 <113035863+RobbiZ98@users.noreply.github.com>

### CODE REVIEW

File: runtime/lua/vim/filetype.lua [hunk 1]
```diff
@@ -466,6 +466,7 @@ local extension = {
   glsl = 'glsl',
   gn = 'gn',
   gni = 'gn',
+  gnuplot = 'gnuplot',
```
Comment (zeertzjq) [inline runtime/lua/vim/filetype.lua:469]: Also need to remove line 1308

Comment (clason) [reply runtime/lua/vim/filetype.lua:469]: No.

Comment (zeertzjq) [reply runtime/lua/vim/filetype.lua:469]: Why? It's duplicate.

Comment (clason) [reply runtime/lua/vim/filetype.lua:469]: No it's not. Test fails if I remove it.

Comment (zeertzjq) [reply runtime/lua/vim/filetype.lua:469]: Oh, that seems to be a much bigger problem.

Comment (clason) [reply runtime/lua/vim/filetype.lua:469]: No, just a standard issue that `extension` doesn't apply to `.extension` files?

Comment (zeertzjq) [reply runtime/lua/vim/filetype.lua:469]: This means Nvim's behavior is different from Vim for many `.extension` files.

Comment (clason) [reply runtime/lua/vim/filetype.lua:469]: Where this matters, we have added the `.extension` files separately. That's the whole point of the filetype tests.

Comment (zeertzjq) [reply runtime/lua/vim/filetype.lua:469]: > Where this matters

Where does this not matter? It should matter for any extension.

Comment (clason) [reply runtime/lua/vim/filetype.lua:469]: In practice. Nobody (sane) is working with literal `.tex` files.

Comment (zeertzjq) [reply runtime/lua/vim/filetype.lua:469]: Yes, but that still matters, doesn't it? I think this needs more discussion in #27976.

**Pull request merged**